src/super_gradients/training/utils/distributed_training_utils.py

```diff
@@ -1,7 +1,51 @@
+import itertools
+from contextlib import contextmanager
 import torch
 from torch.cuda.amp import autocast
 import torch.nn as nn
-import itertools
+
+
+@contextmanager
+def run_once(timeout=None, group=None):
```

Comment:
nice. but I think we already have a simple decorator that does that....
src.super_gradients.common.environment.env_helpers.multi_process_safe
just add @multi_process_safe on top of the download_from_cloud function

Reply:
No it is not. One can use multi_process_safe when there's no need for synchronisation, but that's not the case here.